src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp

```diff
@@ -303,8 +303,18 @@ APPLICATION_INFO::HandleShadowCopy(const ShimOptions& options, IHttpContext& pHt
             }
         }
 
+        int copiedFileCount = 0;
+
         shadowCopyPath = shadowCopyPath / directoryNameStr;
-        HRESULT hr = Environment::CopyToDirectory(physicalPath, shadowCopyPath, options.QueryCleanShadowCopyDirectory(), std::filesystem::canonical(shadowCopyBaseDirectory.path()));
+        LOG_INFOF(L"Copying to shadow copy directory %ls.", shadowCopyPath.c_str());
+
+        // Avoid using canonical for shadowCopyBaseDirectory
+        // It could expand to a network drive, or an expanded link folder path
+        // We already made it an absolute path relative to the physicalPath above
+        HRESULT hr = Environment::CopyToDirectory(physicalPath, shadowCopyPath, options.QueryCleanShadowCopyDirectory(), shadowCopyBaseDirectory.path(), copiedFileCount);
```

Comment:
Can shadowCopyBaseDirectory include '..', '.', etc. that would cause the rfind to return false though, even though they are equivalent paths?

Reply:
No because we make it absolute above. I added a test with a ".." in it to verify.

Reply:
Hm, interesting. The last commit I made in the implementation PR added canonical. https://github.com/dotnet/aspnetcore/pull/28357/commits/dca5ec4a9bdd71763ede9f12e31a81acf341ba9c#diff-300f4ab20a90b601a89471e5c5710667857a6d631e53d5d96d01c39538259122L307. I believe this was to fix an issue where the paths weren't equal with rfind.

Maybe we have the rfind check as well as `|| std::filesystem::equivalent`

Reply:
I don't see why rfind by itself isn't good enough though

Reply:
Are you sure calling absolute will remove '..' and '.'. The docs say absolute will not remove those. https://en.cppreference.com/w/cpp/filesystem/absolute. I'm just very suspicious of this change because this is effectively what I had prior to changing to canonical.

Reply:
I just did some of verification locally on this, and I think maybe the difference was that before I was calling parent_path() on the absolute path [here](https://github.com/dotnet/aspnetcore/pull/28357/commits/dca5ec4a9bdd71763ede9f12e31a81acf341ba9c#diff-300f4ab20a90b601a89471e5c5710667857a6d631e53d5d96d01c39538259122L307).

Change LGTM now.